classes/class-paysoncheckout-for-woocommerce-ajax.php

```diff
@@ -139,6 +139,17 @@ public static function pco_wc_update_checkout() {
 		WC()->cart->calculate_fees();
 		WC()->cart->calculate_totals();
 
+		// Do not update the order if the payson order status is set to readyToShip. This means the order has already been completed.
+		if ( 'readyToShip' === $payson_data['status'] ) {
+			wp_send_json_success(
+				array(
+					'address'   => isset( $payson_data['customer'] ) ? $payson_data['customer'] : null,
+					'changed'   => false,
+					'pco_nonce' => wp_nonce_field( 'woocommerce-process_checkout', 'woocommerce-process-checkout-nonce', true, false ),
+				)
+			);
+		}
+
```

Comment:
This can be done before the calculate_fees and calculate_totals call right? And should reduce how much we do when we know we should exit early.

Also this means the order is already completed. So wont this potentially let a customer continue on the frontend with a order that is actually completed? Should we maybe look at doing something [similar to what we do in KCO](https://github.com/krokedil/klarna-checkout-for-woocommerce/blob/master/classes/class-kco-api.php#L76-L83) and possibly redirect the customer to the thankyou/confirmation page for the order instead here if we can find it in WooCommerce?

Reply:
Yes, good idea! I've implemented the function, and made the suggested changes.